### PR TITLE
Added Integration Tests for fluentd in Operator Environment

### PR DESCRIPTION
**Pull request by hzhao-github**
add automated test to cover the fluentd use case described in

https://oracle.github.io/weblogic-kubernetes-operator/samples/simple/elastic-stack/weblogic-domain/

Passed Jenkins jobs

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3754 (parallel)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3753 (sequential)

### CODE REVIEW

**Review by anpanigr — CHANGES_REQUESTED** [review]
Put the jenkin result on PR description.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+          "uninstallAndVerifyElasticsearch failed with ApiException");
+    }
+
+    if (kibanaParams != null) {
+      logger.info("Uninstall Elasticsearch pod");
+      assertDoesNotThrow(() -> uninstallAndVerifyKibana(kibanaParams),
+          "uninstallAndVerifyKibana failed with ApiException");
+    }
+
+    // delete domain custom resource
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+   * and failed if count is zero.
+   */
+  @Test
+  @DisplayName("Use Fluentd to send log information to Elasticsearch and verify")
+  public void testFluentdQuery() {
+    // Verify that number of logs is not zero and failed if count is zero
+    String regex = ".*count\":(\\d+),.*failed\":(\\d+)";
+    String queryCriteria = "/_count?q=serverName:" + adminServerPodName;
+
+    verifyCountsHitsInSearchResults(queryCriteria, regex, FLUENTD_INDEX_KEY, true);
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+
+    // delete domain custom resource
+    logger.info("Delete domain custom resource in namespace {0}", domainNamespace);
+    assertDoesNotThrow(() -> deleteDomainCustomResource(domainUid, domainNamespace),
+        "deleteDomainCustomResource failed with ApiException");
+    logger.info("Deleted Domain Custom Resource " + domainUid + " from " + domainNamespace);
+  }
+
+  /**
+   * Use log information send by fluentd to Elasticsearch to query logs of serverName=adminServerPodName.
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java]: changed

**Review by anpanigr — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+
+  // constants for creating domain image using model in image
+  private static final String WLS_LOGGING_MODEL_FILE = "model.wlslogging.yaml";
+  private static final String WLS_LOGGING_IMAGE_NAME = "wls-logging-image";
+
+  private static final String FLUENTD_NAME = "fluentd";
+  private static final String FLUENTD_CONFIGMAP_YAML = "fluentd.configmap.elk.yaml";
+  private static final String FLUENTD_IMAGE
+      = "fluent/fluentd-kubernetes-daemonset:v1.3.3-debian-elasticsearch-1.3";
+
```
Comment (marinakog) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java:110]: could you make configurable version based on passed variable for fluent image? so we can update and test against newer versions when available

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+        logger.info("Uninstall Elasticsearch pod");
+        assertDoesNotThrow(() -> uninstallAndVerifyKibana(kibanaParams),
+            "uninstallAndVerifyKibana failed with ApiException");
+      }
+
+      // delete domain custom resource
+      logger.info("Delete domain custom resource in namespace {0}", domainNamespace);
+      assertDoesNotThrow(() -> deleteDomainCustomResource(domainUid, domainNamespace),
+          "deleteDomainCustomResource failed with ApiException");
+      logger.info("Deleted Domain Custom Resource " + domainUid + " from " + domainNamespace);
```
Comment (marinakog) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java:225]: do we need it here? CleanupUtil cleans all generated namespaces

File: integration-tests/src/test/resources/wdt-models/fluentd.configmap.elk.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+apiVersion: v1
```
Comment (marinakog) [inline integration-tests/src/test/resources/wdt-models/fluentd.configmap.elk.yaml:1]: missing copyrights

**Review by marinakog — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java (hunk 4 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java:109]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java (hunk 5 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java]: removed

File: integration-tests/src/test/resources/wdt-models/fluentd.configmap.elk.yaml (hunk 6 above)
Comment (hzhao-github) [reply integration-tests/src/test/resources/wdt-models/fluentd.configmap.elk.yaml:1]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java (hunk 4 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java:109]: I was thinking something like to define var in TestConstants.java like :  public static final String FLUENTD_IMAGE_VERSION = Optional.ofNullable(System.getenv("FLUENTD_IMAGE_VERSION"))
      .orElse("v1.3.3-debian-elasticsearch-1.3"); so you can modify jerking job by adding extra var to switch to different version or change default value

**Review by marinakog — APPROVED** [review]

**Pull request merged**